### DISCUSSION TITLE
ESModule

### DISCUSSION THREAD

**Discussion by tiristoff** [SamWSoftware/serverless-auto-swagger#95] [Q&A]
Hello,

I am working on a project with ts and ESmodule, so I need to have `" type": "module"` on my package.json, but I get the following error.

{"errorMessage":"exports is not defined in ES module scope\nThis file is being treated as an ES module because it has a '.js' file extension and 'C:\\Users\\tiris\\OneDrive\\Documentos\\Proyectos\\sparta-customer-api-serverless\\package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.","errorType":"ReferenceError","stackTrace":["ReferenceError: exports is not defined in ES module scope","This file is being treated as an ES module because it has a '.js' file extension and 'C:\\Users\\tiris\\OneDrive\\Documentos\\Proyectos\\sparta-customer-api-serverless\\package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.","at file:///C:/Users/tiris/OneDrive/Documentos/Proyectos/sparta-customer-api-serverless/swagger/swagger-html.js:2:1","at ModuleJob.run (node:internal/modules/esm/module_job:198:25)","at async Promise.all (index 0)","at async ESMLoader.import (node:internal/modules/esm/loader:385:24)","at async importModuleDynamicallyWrapper (node:internal/vm/module:437:15)","at async _tryAwaitImport (C:\\Users\\tiris\\OneDrive\\Documentos\\Proyectos\\sparta-customer-api-serverless\\node_modules\\serverless-offline\\src\\lambda\\handler-runner\\in-process-runner\\aws-lambda-ric\\UserFunction.js:215:14)","at async _tryRequire (C:\\Users\\tiris\\OneDrive\\Documentos\\Proyectos\\sparta-customer-api-serverless\\node_modules\\serverless-offline\\src\\lambda\\handler-runner\\in-process-runner\\aws-lambda-ric\\UserFunction.js:275:24)","at async _loadUserApp (C:\\Users\\tiris\\OneDrive\\Documentos\\Proyectos\\sparta-customer-api-serverless\\node_modules\\serverless-offline\\src\\lambda\\handler-runner\\in-process-runner\\aws-lambda-ric\\UserFunction.js:304:14)","at async module.exports.load (C:\\Users\\tiris\\OneDrive\\Documentos\\Proyectos\\sparta-customer-api-serverless\\node_modules\\serverless-offline\\src\\lambda\\handler-runner\\in-process-runner\\aws-lambda-ric\\UserFunction.js:341:21)"]}

I tried renaming the files but they are generated again. Any idea on how I can run this plugin with ESModule

**Comment by tiristoff** [comment]
Please can anyone help me with that?

**Reply by bfaulk96** [reply]
I honestly don't know. I haven't looked into splitting this to work with both CommonJS and ESM, and unfortunately don't have the time right now. Always open to reviewing PRs to improve the project, though!

**Comment by tiristoff** [comment]
Thank you very much for your answer
